spark3-extensions/src/main/antlr/org.apache.spark.sql.catalyst.parser.extensions/IcebergSqlExtensions.g4

```diff
@@ -0,0 +1,219 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ * This file is an adaptation of Presto's and Spark's grammar files.
+ */
+
+grammar IcebergSqlExtensions;
```

Comment:
It is debatable whether we should have `Iceberg` in names. We will add MIGRATE and SNAPSHOT here later too. However, CALL will go away once it is available in Spark.

Reply:
I'd also like to add `ALTER TABLE ... [ADD|DROP] PARTITION FIELD ...`

I like having Iceberg in the name because it will be more clear whether something is coming from Iceberg or Spark.

Reply:
It would be great if we can come up with a list of commands that Iceberg would like to add. If possible, I am planning to add `VACUUM`, `OPTIMIZE` and `RESTORE` after this module is added.

Reply:
We have a [design doc](https://docs.google.com/document/d/1Nf8c16R2hj4lSc-4sQg4oiUUV_F4XqZKth1woEo6TN8/edit#) on which commands/stored procedures we are going to support. In addition, we have a [design doc](https://docs.google.com/document/d/1aXo1VzuXxSuqcTzMLSQdnivMVgtLExgDWUFMvWeXRxc/edit#heading=h.41vc5dxcib1o) on how data compaction procedures will look like (i.e. OPTIMIZE). I need to update the latter one.

Reply:
While I think we want to support VACUUM in a way that is close to Redshift, the design of the command requires further consideration. I'd also represent OPTIMIZE as stored procedures.